e2e/cypress/integration/channel/message_shortlinking_spec.js

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+describe('Message', () => {
+    beforeEach(() => {
+        // # Login as "user-1" and go to /
+        cy.apiLogin('user-1');
+
+        cy.visit('/');
+    });
+
+    it('M17451 Channel shortlinking still works when placed in brackets', () => {
+        // # Post shortlink of channel
+        const shortLink = '(~saepe-5)';
+        const longLink = '~doloremque';
+
+        cy.postMessage(shortLink);
+
+        // # Grab element with the long link url and go to the link
+        cy.get('a.mention-link').contains(longLink).click();
+
+        // * verify that the url is the same as what was just clicked on
+        cy.location('pathname').should('contain', 'saepe-5');
```

Comment:
I suggest to test with longer path:
```suggestion
        cy.location('pathname').should('contain', '/ad-1/channels/saepe-5');
```

Reply:
got it! not sure how the  ad-# will change but this will work for most numbers. is this okay?
```js
        const URLRegularEx = /\/ad-[0-9]+\/channels\/saepe-5/;
        cy.location('pathname').should('match', URLRegularEx);
```

Reply:
I don't think you need the regex, the 'ad-1' is the one of the default team's that get's created, so the number isn't really important and rather just a part of the name itself.